src/lib/repositories/impl_no_sql/order_status_history_repository_impl.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime
+
+from src.constants.audit import Status, InternalUsers
+from src.core.mongo_engine_config import mongo_engine_connection
+from src.lib.entities.mongo_engine_odm_mapping import OrderStatusHistory
+from src.lib.repositories.order_status_history_repository import (
+    OrderStatusHistoryRepository,
+)
+
+
+class OrderStatusHistoryRepositoryImpl(OrderStatusHistoryRepository):
+    def __init__(self):
+        mongo_engine_connection()
+
+    def add(self, order_status_history):
+        order_status_history.created_date = datetime.now()
+        order_status_history.updated_by = order_status_history.created_by
+        order_status_history.updated_date = order_status_history.created_date
+        order_status_history.save()
+
+    def get_by_id(self, order_status_history_id):
+        order_status_history = OrderStatusHistory.objects.get(
+            id=order_status_history_id, entity_status=Status.ACTIVE.value
```

Comment:
I think this and the others are wrong. You need to use mongo id version instead.

Reply:
done.